apis/interfaces.go

```diff
@@ -66,3 +66,10 @@ type Listable interface {
 // The webhook functionality for this has been turned down, which is why this
 // interface is empty.
 type Annotatable interface{}
+
+// HasSpec indicates that a particular type has a specification information
+// and that information is retrievable.
+type HasSpec interface {
```

Comment:
Quoting from https://github.com/knative/eventing/pull/1293

> * The reason for `HasSpec` interface is, in the CRD using the `spec` field for specification of a resource is a convention but following also would work if the operator is designed to read the specs from `LeSpecification` field:

```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
spec:
  ...
  validation:
    openAPIV3Schema:
      properties:
        LeSpecification:
          properties:
            ...
```

Reply:
This should be `GetUntypedSpec` or something.  Otherwise, implementations cannot use `GetSpec` to return their typed Spec object, which I was literally just recommending for the Ingress work in Serving.

Reply:
+1